mm2src/coins/utxo/rpc_clients/electrum_rpc/connection.rs

```diff
@@ -251,7 +251,7 @@ impl ElectrumConnection {
         self.responses
             .lock()
             .unwrap()
-            .insert(rpc_id, req_tx, Duration::from_secs_f64(timeout));
+            .insert_expirable_unchecked(rpc_id, req_tx, Duration::from_secs_f64(timeout));
```

Comment:
the only insert being unchecked basically makes this map susceptible to memory leaks. this acts pretty much as a normal hashmap at this point.

you can instead make it a checked insert but use a high tick cap.

there are other never-checked maps also introduced in this PR.

Reply:
> the only insert being unchecked basically makes this map susceptible to memory leaks. this acts pretty much similar (with an *) to a normal hashmap at this point.

> you can instead make it a checked insert but use a high tick cap.

We don't need to include expiration overhead on inserts on electrum requests. The check is performed at end of `process_electrum_response` which should be enough.

> there are other never-checked maps also introduced in this PR.

Can you mark them? I tried to minimize the expiration overhead in various places where it's unnecessary.

Reply:
> We don't need to include expiration overhead on inserts on electrum requests. The check is performed at end of process_electrum_response which should be enough.

i can't find no check for outdated entries in `process_electrum_response`!
you aren't talking about the `.remove` right? `.remove` will not help with requests that we got no responses for, these request handles will reside in the timed map forever (until the connection terminates*).